pkg/release/release.go

```diff
@@ -184,10 +185,14 @@ func (r *Release) EnsureDeleted(ctx context.Context, name string, conditions ...
 }
 
 // EnsureInstalled makes sure the release is installed and all conditions are
-// met.
+// met. If release name ends with "-operator" suffix it also selects
+// a "app=${name}" pod and streams it logs to the ./logs directory.
 //
 // NOTE: It does not update the release if it already exists.
 func (r *Release) EnsureInstalled(ctx context.Context, name string, version Version, values string, conditions ...func() error) error {
+	var err error
+	isOperator := strings.HasSuffix(name, "-operator")
```

Comment:
Not sure if this is the best approach on earth. But previous one was also based on heavy assumptions.

Reply:
What is the benefit of doing this insteal of `InstallOperator` and using `EnsureInstalled` inside?

Reply:
To me this does not seem intuitive, when I call `InstallOperator` I expect something different to happen than just calling `Install`. Here there is no indication that something different is happening behind the scenes.

Reply:
I want to have idempotent version of installing an operator. Also having consistency is nice. What's the alternative? `EnsureOperatorInstalled`? 

To make the logging nice I would write exactly like that:

```
func (r *Release) EnsureInstalled(ctx context.Context, name string, version Version, values string, conditions ...func() error) error {
	err := r.ensureInstalled(ctx, name, version, values, false, conditions...)
	if err != nil {
		return microerror.Mask(err)
	}
	
	return nil
}

func (r *Release) EnsureOperatorInstalled(ctx context.Context, name string, version Version, values string, conditions ...func() error) error {
	err := r.ensureInstalled(ctx, name, version, values, true, conditions...)
	if err != nil {
		return microerror.Mask(err)
	}
	
	return nil
}

func (r *Release) ensureInstalled(ctx context.Context, name string, version Version, values string, isOperator bool, conditions ...func() error) error {
}
```

Is that better?

Reply:
I don't get why you want to mix both things in `ensureInstalled`. `EnsureOperatorInstalled` adds additional functionality around the base functionality of `EnsureInstalled`. 

May solution would be to simply replace this line here with `EnsureInstalled`:
https://github.com/giantswarm/e2e-harness/blob/9f97abe68c8fb2ce7a50886098d4d4d24d187583/pkg/release/release.go#L283

Reply:
I'd like to setup logging before we start waiting for CRD. Otherwise we can miss some error.

Reply:
I mean error during CRD creation in the operator. That was a flaw of previous implementation.

Reply:
Okay I see your point. I still would not want two functions doing the same thing in different ways (`InstallOperator` and `EnsureInstalled`). So if that is a problem then let's fix it for everything.

Reply:
I'd like to get rid of `InstallOperator` completely.
I'd also like to remove `conditions` form `Install` and `Upadate` (where they aren't used btw) and make those methods lean. We should use only `Ensure*` implementations unless we have very special cases. In such cases I think we can use conditions explicitly.

Reply:
> I'd like to get rid of InstallOperator completely.

Maybe it's a good idea to do this in this PR?

Reply:
I agree, let's remove `InstallOperator` in this PR then and vendor this into our major operators. If everything goes well, then we can merge here.

Reply:
https://github.com/giantswarm/e2e-harness/pull/155#discussion_r228425284